keras_cv/layers/spatial_pyramid.py

```diff
@@ -133,9 +112,13 @@ def build(self, input_shape):
                 tf.keras.layers.GlobalAveragePooling2D(),
                 tf.keras.layers.Reshape((1, 1, channels)),
                 tf.keras.layers.Conv2D(
-                    filters=self.num_channels, kernel_size=(1, 1), use_bias=False
+                    filters=self.num_channels,
+                    kernel_size=(1, 1),
+                    use_bias=False,
                 ),
-                tf.keras.layers.BatchNormalization(),
+                tf.keras.layers.experimental.SyncBatchNormalization(),
+                # TODO(tanzhenyu): File a bug to Keras for NAN issue.
```

Comment:
maybe link to https://github.com/keras-team/keras/issues/17204 or https://github.com/tensorflow/tensorflow/issues/48845?

Reply:
@tanzhenyu Can you quickly verify if you can reproduce it with `drop_remainder=True`?

Reply:
Looks like the problem goes away. (I'm running the script to see if quality wise it makes any difference)